### ISSUE TITLE
Fail to create release using non-latest commit isa with "GitHub release failed with status: 403"

### ISSUE THREAD

**Issue by winni0713** [softprops/action-gh-release#411]
I am only able to create release using latest commit sha. If I provide older commit sha, "GitHub release failed with status: 403" error is hit.

![image](https://github.com/softprops/action-gh-release/assets/12772076/2f04b227-122e-4ed4-b2a3-bc998d98deb1)

`name: pre-release

on:
  workflow_dispatch:
    inputs:
        release_name:
          required: true
        commitSHA:
          required: true
          
permissions: write-all
  #contents: write

jobs:
  release:
    name: release
    runs-on: ubuntu-latest

    steps:

      - name: create release
        id: create_release
        uses: softprops/action-gh-release@v1
        with:
          draft: false
          prerelease: true
          name: ${{ inputs.release_name }}
          tag_name: ${{ inputs.release_name }}
          target_commitish: ${{ inputs.commitSHA }}
          #body_path: CHANGELOG.md

        env:
          GITHUB_TOKEN: ${{ github.token }}
          
    `

**Comment by ForNeVeR** [comment]
Likely same as #400?

**Comment by winni0713** [comment]
> Likely same as #400?

No, i have added the write permission in yml file according to the solution in this issue. So using a few recent commit is fine to create release, but older one will hit error.

**Comment by chrisohaver** [comment]
I am seeing the same error.  All workflows have read/write. And I have also added write permission to the workflow yml. I am using v2 of the workflow action. `softprops/action-gh-release@v2`

@winni0713, what commit is working for you?

**Comment by chrisohaver** [comment]
Using latest release I am seeing the same failure.
`softprops/action-gh-release@9d7c94cfd0a1f3ed45544c887983e9fa900f0564`

I'm not sure if it matters, but in or case a tag with the same name at the same commit had existed before but was deleted via the GitHub UI.  Could this prevent the creation of the tag now?

**Comment by winni0713** [comment]
@chrisohaver , i was using v1 previously and normally the latest 2 commit can create the release successfully. But it is strange that it is not reproducible in my another repo. Previous commit can still be used.

**Comment by chenrui333** [comment]
I could not reproduce or isolate this on current master and the latest 2.5.2 release in the current regression sweep, so I am closing it to keep the backlog focused on confirmed current bugs. If you are still seeing a different current failure, please open a new issue with a minimal repro or sanitized workflow snippet and the failing logs.

**Comment by chenrui333** [comment]
Closing this out based on current testing with the latest code and the merged docs update in #758.

The latest repro still hits `403 Resource not accessible by integration` when `github.token` is used to create a release/tag against an older commit. A direct ref-creation diagnostic against the same target SHA also fails with the same permission boundary, so this currently points to token permissions rather than an action-side bug.

If you still see a different behavior or have a minimal repro that points to an action-side issue, please open a new bug report with the current issue template.
